### DISCUSSION TITLE
Binary compatibility & version

### DISCUSSION THREAD

**Discussion by SylvainCorlay** [nlohmann/json#2428] [Q&A]
I am working on the packaging for conda, which is a binary package manager, of several libraries that depend on nlohmann_json. What is the expected binary compatibility binaries built with different versions of nlohmann_json, depending on how these versions differ?

For example, if `B` depends on `A`. Last time `A` was built, it was built with nlohmann_json `3.7.0`.

Can `B` link with `A` and depend on nlohann_json `3.7.x`? (it seems to be the case so far, but I don't know if it will always be)
Can `B` link with `A` and depend on nlohann_json `3.x.x`? (It seems the latter is not true)

The binary compatibility of versions differing by the patch version number is what I have assumed so far. This impacts the `run_exports` section in the package metadata of conda-forge.

**Comment by nlohmann** [comment]
> What is the expected binary compatibility binaries built with different versions of nlohmann_json, depending on how these versions differ?

I have no expectation on this. ABI stability was not a concern so far, so I would guess that there is no binary compatibility.
